ios/Runner/AppDelegate.swift

```diff
@@ -15,18 +15,31 @@ import FirebaseCore
         
         FirebaseApp.configure()
         
+        var modifiedLaunchOptions = launchOptions
+        
+        if let remoteNotification = launchOptions?[UIApplication.LaunchOptionsKey.remoteNotification] as? [String: Any],
```

Comment:
I think this PR is fixing deep links for the sample app? 

This code is coupled tightly to the CIO SDK since it's parsing the notification. 

Is there a more generic solution that would work for any deep link? Not just a deep link from CIO SDK?

Reply:
Yes, this is tightly coupled to the SDK because the problem I was trying to solve here is actually with Rich Push Notification link only.

Without this fix (in `main` branch), if I open `amiapp-flutter://ios/settings` using notes app when sample app is in killed state, it lands me to the settings page which is the correct behavior. But if I tap on rich push with same link, it lands me to home page only and doesn't navigate to settings page when app is in killed state.

After this fix, in both cases, I'm navigated to the settings page which is the correct behavior.

Reply:
Have you tried [setting up the iOS flutter app with deep links like our docs suggest](https://customer.io/docs/sdk/ios/push/#deep-links)? 

The solution our iOS docs suggest is more generic and not tightly coupled to our SDK so it's a preferred solution for customers. 

If you have tried this and didn't have any luck, could you share what issues you encountered so we could fix it in a future PR? Then we can merge this PR as-is and improve it in the future.

Reply:
Thanks a lot for sharing this @levibostian. I didn't try initializing SDK again which actually did the trick and made the changes a lot simpler 🎉  I have reverted the changes and added only SDK initialization now.